psydac/fem/splines.py

```diff
@@ -171,14 +171,17 @@ def init_interpolation( self, dtype=float ):
         Greville points.
 
         """
-        imat = collocation_matrix(
-            knots    = self.knots,
-            degree   = self.degree,
-            periodic = self.periodic,
-            normalization = self.basis,
-            xgrid    = self.greville,
-            multiplicity = self.multiplicity
-        )
+        if self.greville.size == 1:
+            imat = np.ones((1, 1), dtype=float)
+        else:
+            imat = collocation_matrix(
+                knots    = self.knots,
+                degree   = self.degree,
+                periodic = self.periodic,
+                normalization = self.basis,
+                xgrid    = self.greville,
+                multiplicity = self.multiplicity
+            )
```

Comment:
I suggest that this case be handled inside the function `collocation_matrix` itself.

Reply:
Done in df5ed36